File: frontend/src/services/cronogramaService.ts
import api from './api';

export type Cronograma = {
  id?: string;
  metaId: string;
  dataPrevista?: string; // ISO date
  atividade?: string;
  createTime?: string;
  updateTime?: string;
};

const base = '/cronograma';

export const listByMeta = async (metaId: string): Promise<Cronograma[]> => {
  const res = await api.get(`${base}/meta/${metaId}`);
  return res.data;
};

export const create = async (payload: Cronograma): Promise<Cronograma> => {
  const res = await api.post(base, payload);
  return res.data;
};

export const update = async (id: string, payload: Cronograma): Promise<Cronograma> => {
  const res = await api.put(`${base}/${id}`, payload);
  return res.data;
};

export const remove = async (id: string): Promise<void> => {
  await api.delete(`${base}/${id}`);
};

export default { listByMeta, create, update, remove };

File: src/main/java/org/acme/service/CronogramaService.java
package org.acme.service;

import org.acme.dto.CronogramaDTO;
import org.acme.entity.Cronograma;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.transaction.Transactional;
import java.util.List;
import java.util.stream.Collectors;

@ApplicationScoped
public class CronogramaService {

    public List<CronogramaDTO> listByMeta(String metaId) {
        return Cronograma.find("metaId", metaId).stream()
                .map(o -> (Cronograma)o)
                .map(c -> toDto(c))
                .collect(Collectors.toList());
    }

    @Transactional
    public CronogramaDTO create(CronogramaDTO dto) {
        Cronograma c = new Cronograma();
        c.metaId = dto.metaId();
        c.dataPrevista = dto.dataPrevista();
        c.atividade = dto.atividade();
        c.persist();
        return toDto(c);
    }

    @Transactional
    public CronogramaDTO update(String id, CronogramaDTO dto) {
        Cronograma c = Cronograma.findById(id);
        if (c == null) return null;
        c.dataPrevista = dto.dataPrevista();
        c.atividade = dto.atividade();
        return toDto(c);
    }

    @Transactional
    public boolean delete(String id) {
        Cronograma c = Cronograma.findById(id);
        if (c == null) return false;
        c.delete();
        return true;
    }

    private CronogramaDTO toDto(Cronograma c) {
        return new CronogramaDTO(c.id, c.metaId, c.dataPrevista, c.atividade, c.createTime, c.updateTime);
    }
}
